src/Utils.hpp

```diff
@@ -68,6 +72,23 @@ inline std::shared_ptr<BaseIO> createIO(const std::string& type,
   }
 }
 
+inline void convertFloatToInt16LE(const float* source,
+                                  void* dest,
+                                  int numSamples)
+{
+  auto maxVal = static_cast<double>(0x7fff);
+  auto intData = static_cast<char*>(dest);
+
+  for (int i = 0; i < numSamples; ++i) {
+    auto clampedValue = std::clamp(maxVal * source[i], -maxVal, maxVal);
+    auto intValue =
+        static_cast<uint16_t>(static_cast<int16_t>(std::round(clampedValue)));
+    intValue = boost::endian::native_to_little(intValue);
+    *reinterpret_cast<uint16_t*>(intData) = intValue;
+    intData += 2;  // destBytesPerSample is always 2
```

Comment:
Is the data being converted to Int16 or Uint16? The function name implies Int16 but it seems the code is for UInt16. If it's UINT, then I would propose to change the name of the function accordingly. Also, please add a docstring for the function.